### ISSUE TITLE
Incorrect error location

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2#3470]
This error location is not helpful, nor correct, I believe.
```m2
f = (a,b) -> (
    --
    a -- <<< line 3
    --
    )

g = x -> (
    --
    f x
    --
    )

end
restart
needs "test.m2"

g 1 -- test.m2:3:4:(3): error: expected 2 arguments but got 1
```
cc: @pzinn

**Comment by mahrud** [comment]
Real world example: https://github.com/Macaulay2/M2/actions/runs/10807529614/job/29978522685?pr=3469

**Comment by pzinn** [comment]
Weird. Works fine on Macaulay2Web, though:
```
i1 : needs "errtst.m2"

i2 : g 1

errtst.m2:1:10:(3): error: expected 2 arguments but got 1
```
I'll investigate.

**Comment by pzinn** [comment]
The bug was introduced by your commit 73430ac024339eb9780d9d418155d1378237acf8.

**Comment by pzinn** [comment]
err wrong link let me try again:

**Comment by pzinn** [comment]
the bug was introduced in 1a2a99216ecc54afe4ec7691ac4241dc1861107f.

**Comment by pzinn** [comment]
this is the offending line in `codePosition`
```
     is f:functionCode do codePosition(f.body)
```
instead of 
```
     is f:functionCode do f.position
```
not sure how that happened.

**Comment by pzinn** [comment]
@mahrud is that a change you made? if not I'll PR the one-line fix.

**Comment by mahrud** [comment]
Nope, changes I made were committed under my name only. The rest are only pieces of your big commit.
